### PR TITLE
[BE-866] Adding helper functions for paypal checkout

### PR DESCRIPTION
**Pull request by inai-harish**
## Description:

- Adds helper functions for making REST API calls to PayPal checkout APIs.
- Adds response handling functions (Used other gateway integrations as a reference)

### Sample responses:

Response from PayPal:
```
{
    "id": "26715072L61245648",
    "status": "CREATED",
    "links": [
        {
            "href": "https://api.sandbox.paypal.com/v2/checkout/orders/26715072L61245648",
            "rel": "self",
            "method": "GET"
        }
    ]
}
```
converted to active-merchant's response as 
```
{
   "params":{
      "id":"7W1354689B616405R",
      "status":"CREATED",
      "links":[
         {
            "href":"https://api.sandbox.paypal.com/v2/checkout/orders/7W1354689B616405R",
            "rel":"self",
            "method":"GET"
         }
      ]
   },
   "message":"CREATED",
   "success":true,
   "test":"true",
   "authorization":"7W1354689B616405R",
   "fraud_review":null,
   "error_code":null,
   "emv_authorization":null,
   "network_transaction_id":null,
   "avs_result":{
      "code":null,
      "message":null,
      "street_match":null,
      "postal_match":null
   },
   "cvv_result":{
      "code":null,
      "message":null
   }
}
```

### CODE REVIEW

File: lib/active_merchant/billing/gateways/paypal/paypal_checkout_common.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  module Billing
+    module PaypalCheckoutCommon
+
+      URLS = {
+        :test_url     => "https://api.sandbox.paypal.com",
+        :live_url     => "https://api.paypal.com"
+      }
+
+      ALLOWED_INTENT              = %w(CAPTURE AUTHORIZE).freeze
+      ALLOWED_PAYMENT_TYPE        = %w(ONE_TIME RECURRING UNSCHEDULED).freeze
```
Comment (inai-shreyas) [inline lib/active_merchant/billing/gateways/paypal/paypal_checkout_common.rb:11]: what does UNSCHEDULED do?

**Review by inai-shreyas — COMMENTED** [review]
https://github.com/inaitech/active_merchant/blob/master/lib/active_merchant/billing/gateways/paypal/paypal_common_api.rb
This one has standard error code mapping as well. can we add that as well to this?

File: lib/active_merchant/billing/gateways/paypal/paypal_checkout_common.rb (hunk 1 above)
Comment (inai-harish) [reply lib/active_merchant/billing/gateways/paypal/paypal_checkout_common.rb]: Removed them as the recurring API supported by PayPal is not used in our integration.

**Comment by inai-harish** [comment]
> https://github.com/inaitech/active_merchant/blob/master/lib/active_merchant/billing/gateways/paypal/paypal_common_api.rb
> This one has standard error code mapping as well. can we add that as well to this?

@inai-shreyas Cannot find any supporting doc for handling these errors for the /v2 APIs with which we are integrating except for  https://developer.paypal.com/docs/business/checkout/reference/card-decline-errors/. Will add the response handlings in the subsequent PRs while integrating with the PayPal APIs.

File: lib/active_merchant/billing/gateways/paypal/paypal_checkout_common.rb (hunk 1 above)
Comment (inai-shreyas) [reply lib/active_merchant/billing/gateways/paypal/paypal_checkout_common.rb]: is it ONE_TIME by default?

**Review by inai-shreyas — COMMENTED** [review]
LGTM

**Pull request merged**